requirements/basic.txt

```diff
@@ -12,3 +12,4 @@ pydantic==2.*
 PyYAML
 dask
 toml
+vtk
```

Comment:
The installation package for vtk is about 100mb. It doesn't seem to have any installation issues (just `pip install vtk`) on different systems, which is probably not surprising since it's a very mature package (especially its C++ version). But yeah, I think it probably makes sense to treat it in the same way as `trimesh`, especially since it would be required only if you are working with heat solver or in rare cases when the user wants to import unstructured data from other solvers.

Reply:
The idea of the `basic` requirements is that you should be able to load and validate simulations, even if you can't do anything else. Generally, I do think it would be best if we can encapsulate vtk imports in a `VTK_AVAILABLE` kind of handling similar to [trimesh](https://github.com/flexcompute/tidy3d/blob/f60bf47f84264bc622e548fa41dfc344fb85ceaf/tidy3d/components/geometry/mesh.py#L21-L26). Do you think it can be possible to load simulations with unstructured data if vtk is not installed? Maybe some validators would have to be skipped `if not VTK_AVAILABLE`?

Reply:
yeah, that should be possible. VTK is really needed only for performing operations like slicing/clipping/interpolating

Reply:
Switched `vtk` to be an optional package. Similar to `trimesh`, functions that require `vtk` will raise an error if it's not available. And similar to `trimesh`, it is attempted to be imported in `types.py`. So, in the rest of the code `from .types import vtk` and `if vtk == None:` is used. Also managed to add a test that checks that if `vtk` is not available instances can still be created and validated but most of functionality is disabled. I did it in the following way:
- `test_datasets.py` contains tests that check behavior for both when `vtk` is and is not available via `if vtk is None:`. However, running `pytest test_datasets.py` will check only one side of that depending whether `vtk` is installed in the python environment
- so, there is a separate file `_test_datasets_no_vtk.py` that imports the same tests but prepend them with a monkeypatch fixture that mocks `vtk` is not available. So, running `pytest _test_datasets_no_vtk.py` checks that everything works as expected when `vtk` is not installed. It has to be run as a separate pytest session otherwise `vtk` gets imported in other tests and remains loaded (hence, changes to `test_local.py` and `tox.ini`)